### ISSUE TITLE
Mentoring locked, because of other mentor?

### ISSUE THREAD

**Issue by dviererbe** [exercism/exercism#6532]
Today I started a mentor session. I typed a long message responding to the questions asked.
When wanted to send my initial message I got the message that the user was locked because of an other Mentor.

I thought it was a bug; returned to the mentor dashboard and the user wasn't listed anymore.

**Maybe relevant Information:**
I submitted a solution for the same problem on the Java Track (based on my C# submission), because I do much more C# programming to verify that my answer is correct and avoid giving bad advise. I don't think that this is relevant, because I already did this in the past. 

**Relevant Information:**
My Username: _dviererbe_
Username of user to mentor: _Heat_
Track: _Java_
Exercise: _Queen Attack_

**UX learning:**
To put it simply, this was not a great experience. It took a long walk to keep calm and reflect on the situation instead of walking away from mentoring (or the platform) forever.
Whatever the reason for this error was. There should be a better visible warning or a mechanism to avoid the situation entirely.

**Comment by iHiD** [comment]
Hi @dviererbe - thanks for the message. Really sorry that you had this experience. I've just been through the logs. What actually happen was that the user removed their request for mentoring just before you published, so then when we can to post your feedback, there was no longer a mentor request that we could fulfil. 

Things we could improve:
1. A better error message that told you that's what happened
2. A websocket notification to you that the student had removed the mentoring request so you could stop typing your response.

However, the way the timings worked on this, even if we'd done those things, you'd have done most of the work by the time this happened, so you'd have had a rubbish experience still.

I therefore suggest two more improvements
3. Add a note to a user's solution that a mentor is giving them feedback
4. If they do cancel, warn the user that a mentor has already started giving feedback, and encourage them not to cancel their request.

Is there anything else we could do to make the UX work better here?

Please let me know, then I'll close this issue and copy the conclusions over to https://github.com/exercism/exercism/issues/6056 which is an existing conversation about this issue.

**Comment by dviererbe** [comment]
Well, that was really an unfortunate series of events. Thanks for the explaination!

All 4 ideas are really good and I don't have anything to add. Thanks for your effort ❤️

**Comment by iHiD** [comment]
OK great. I'll copy stuff over. Sorry again and thanks for being so kind/understanding about it.
